### ISSUE TITLE
Unicode with facetgrid still doesnt work on version 0.7 or 0.8dev

### ISSUE THREAD

**Issue by moosekaka** [mwaskom/seaborn#905]
I am still getting this error  using the MWE from issue [#719](https://github.com/mwaskom/seaborn/issues/719)

```
File "<ipython-input-7-b758835010eb>", line 11, in <module>
    g.map(plt.scatter, 'x', 'y')
...
UnicodeEncodeError: 'ascii' codec can't encode characters in position 0-4: ordinal not in range(128)
```

I am running Python v2.7.11, Anaconda2 and Seaborn 0.7 and also tried 0.8dev

**Comment by mwaskom** [comment]
cc @gkunter

**Comment by mwaskom** [comment]
@moosekaka it would be more helpful to fully reproduce the example that is failing here...

**Comment by moosekaka** [comment]
```
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline
data = sns.load_dataset('anscombe')
data['dataset_unicode'] = data.dataset.map(lambda x: u'ÁéÍóÚ {0}'.format(x))
g = sns.FacetGrid(data, col='dataset_unicode')
g.map(plt.scatter, 'x', 'y')
```

output error:
```
Traceback (most recent call last):

  File "<ipython-input-8-4da69c8fa7a1>", line 9, in <module>
    g.map(plt.scatter, 'x', 'y')

  File "C:\Users\sweel_rafelski\Anaconda2\lib\site-packages\seaborn\axisgrid.py", line 730, in map
    self._finalize_grid(args[:2])

  File "C:\Users\sweel_rafelski\Anaconda2\lib\site-packages\seaborn\axisgrid.py", line 820, in _finalize_grid
    self.set_titles()

  File "C:\Users\sweel_rafelski\Anaconda2\lib\site-packages\seaborn\axisgrid.py", line 962, in set_titles
    title = template.format(**args)

UnicodeEncodeError: 'ascii' codec can't encode characters in position 0-4: ordinal not in range(128)
```

**Comment by mwaskom** [comment]
FWIW I cannot reproduce this error on Python 2.7, seaborn 0.8.dev, and matplotlib 1.5.0.

**Comment by ajasja** [comment]
@mwaskom 
Yes, this seems fixed in 0.8 dev, but is reproducible in 0.7. Here is the test notebook I used:
https://gist.github.com/ajasja/40378cb2514e510c48d66ae330137776

Probably this can be closed.

**Comment by ajasja** [comment]
@mwaskom 
Although there is a bug in categorical.py, that can be fixed by just putting an u in line 1502.

Should I do a one char pull request?

Notebook is here:
https://gist.github.com/ajasja/953c97a3376edd09e625653527319831

[BarvniPasovi.xlsx](https://github.com/mwaskom/seaborn/files/306658/BarvniPasovi.xlsx)
